### ISSUE TITLE
Cannot access TiDB 4.0 pd dashboard

### ISSUE THREAD

**Issue by kolbe** [pingcap/tidb-operator#2273]
## Bug Report

I deployed a cluster using TiDB 4.0.0-rc and was unable to access the /dashboard/ provided by the pd service.

After port-forwarding port 2379 to the pd service, the CluserIP sent me to a non-leader pd pod, and it redirected me to the leader:

```
<a href="https://kolbe-test-cluster-pd-1.kolbe-test-cluster-pd-peer.tidb-cluster.svc:2379/dashboard/">Temporary Redirect</a>.
```

Well, I of course can't resolve that hostname outside of the k8s cluster, so the redirect doesn't work.

I think we probably need to implement a dashboard proxy inside k8s that can handle redirecting to the pd leader and can restrict access to /dashboard/ so the entire pd API isn't exposed.

**Comment by kolbe** [comment]
After forwarding the port to the "correct" pd pod (rather than to the service), I do get the login page, but login fails if the backend cluster uses TLS for connections between TiDB Server and its MySQL-compatible clients.

I get a "Sign in failed: error.api.user.signin.other" error shown on the webpage, but the raw HTTP results show more detail:

```
"authenticate failed, cause: error.api.user.signin.other: x509: certificate is valid for kolbe-test-cluster-tidb, kolbe-test-cluster-tidb.tidb-cluster, kolbe-test-cluster-tidb.tidb-cluster.svc, *.kolbe-test-cluster-tidb, *.kolbe-test-cluster-tidb.tidb-cluster, *.kolbe-test-cluster-tidb.tidb-cluster.svc, not kolbe-test-cluster-tidb-0.kolbe-test-cluster-tidb-peer.tidb-cluster.svc"
```

PD is trying to connect to the backend tidb instance using its pod hostname, which is not included in the SAN of the TLS certificate, so TLS cannot be verified. PD should probably connect to the TiDB *service* endpoint instead of to a specific *pod* endpoint.

**Comment by kolbe** [comment]
https://github.com/pingcap/pd/issues/2375

**Comment by DanielZhangQD** [comment]
This may be supported together with https://github.com/pingcap/tidb-operator/issues/2276

**Comment by Yisaer** [comment]
It seems the problem has been solved in pingcap/docs-tidb-operator#215 as mentioned in https://github.com/pingcap/pd/issues/2375.

Feel free to reopen issue if you have any further question.
